agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/telemetry/TelemetryClient.java

```diff
@@ -155,27 +158,26 @@ public void trackAsync(TelemetryItem telemetry) {
     MonitorDomain data = telemetry.getData().getBaseData();
     if (data instanceof MetricsData) {
       MetricsData metricsData = (MetricsData) data;
-      List<MetricDataPoint> filteredPoints =
-          metricsData.getMetrics().stream()
-              .filter(
-                  point -> {
-                    String metricName = point.getName();
-                    if (nonFilterableMetricNames.contains(metricName)) {
-                      return true;
-                    }
-                    for (MetricFilter metricFilter : metricFilters) {
-                      if (!metricFilter.matches(metricName)) {
-                        return false;
-                      }
-                    }
-                    return true;
-                  })
-              .collect(Collectors.toList());
-
-      if (filteredPoints.isEmpty()) {
+      if (metricsData.getMetrics().isEmpty()) {
+        // this is unexpected
+        logger.debug("MetricsData has no metric point");
```

Comment:
should it throw an exception if it's unexpected?

Reply:
I was being conservative, but feel pretty confident, so bumped it up to an exception